### ISSUE TITLE
DropPath and RoPE augmentations enabled during Utonia linear probing?

### ISSUE THREAD

**Issue by ChristianSchott** [Pointcept/Pointcept#606]
Hello,

I am currently trying to replicate the linear probing results from the paper with the pre-trained Utonia checkpoint on ScanNet.

Currently my best val is at around 75.3% mIoU, 86.5% mAcc, and 90.7% after 8 epochs. 

<img width="1883" height="769" alt="Image" src="https://github.com/user-attachments/assets/c0930e9f-eb9c-492f-8ef6-6c5bf09c2f52" />

So still ~1% off from the results in the paper, but the head may not be fully converged yet + I have divided bs/3 and lr/3 as I am training on a single A40.. 

However, to me it seemed a bit strange how "jumpy" the validation curves look (or is that expected?).

While looking for potential causes I noticed, that `drop_path=0.3`, `jitter_coords=1.1` and `rescale_coords=1.2` for all lin/dec/ft configs.
`DefaultSegmentorV2` also only freezes the backbone parameters, without setting the backbone to `.eval()`, so the `training`-state will be `true` for the frozen backbone, causing `drop_path` and rope augentations to remain active during the training. Is this intentional?

Are the configs in the repo, the exact ones used during the evals for the paper, or were they created post-hoc?
Also do you have a rough guidance after how many epochs I should expect full convergence for linear probing?

Thank you and best regards
Christian

**Comment by Yujia-Zhang0913** [comment]
Hi,
Thanks for your interest in Utonia. The model should be `.eval()` when freezing the encoder. I'll have a check on this issue in next commit.

The configs are the exactly the same configuration for results in report. The learning rate and batch size will have influence on the results. Maybe you can do grid search for the suitable learning rate.

Feel free to contact me for any further questions!

**Comment by ChristianSchott** [comment]
Hi, 

unless I am missing something here, `freeze_encoder` of Utonia is set to `False`, and even if set to `True` it would only set `requires_grad=False`, without setting the encoder to `.eval()`.
https://github.com/Pointcept/Pointcept/blob/d7272257058337969292865e0892b91d2f362b6a/configs/utonia/semseg-utonia-v1m1-0a-scannet-lin.py#L40-L51

`freeze_backbone` of `DefaultSegmentorV2` is set to `True` in the config, however that also only disables gradients without setting `.eval()`.
https://github.com/Pointcept/Pointcept/blob/d7272257058337969292865e0892b91d2f362b6a/pointcept/models/default.py#L40-L62

If I remember correctly, I even logged `self.training` of the backbone and it returned `true`.. 

So I guess linear probing results of Utonia could be even better than reported in the paper? Though, hard for me to verify without running the experiment with full batch size.. maybe I find some time to try this on the cluster..

**Comment by ChristianSchott** [comment]
Hi again,

I have just randomly noticed that indoor-scaling is not set in the lin/dec/f.t. configs for ScanNet++ (all others seem fine, as far as I have checked).

This could also explain the comparatively low segmentation results of Utonia on ScanNet++.

<img width="809" height="283" alt="Image" src="https://github.com/user-attachments/assets/e44e895f-555d-496b-ac7a-8d92716c7af4" />

https://github.com/Pointcept/Pointcept/blob/9f37497e4f3005c90bbbe7221b86439c29d60611/configs/utonia/semseg-utonia-v1m1-2a-scannetpp-lin.py#L81-L92

Note that there is no `dict(type="RandomScale", scale=[0.5, 0.5])` like for other indoor datasets.

@Yujia-Zhang0913

**Comment by ChristianSchott** [comment]
Regarding the original topic of this issue:

I just wanted to note that setting `self.backbone.eval()` in the `DefaultSegmentorV2` init may not be enough by itself, as the `Trainer`

https://github.com/Pointcept/Pointcept/blob/9f37497e4f3005c90bbbe7221b86439c29d60611/pointcept/engines/train.py#L157-L166

resets `self.model.train()` each epoch.
So I guess the `DefaultSegmentorV2` could become a `PointModel` (+`ModelHook`) and set `self.backbone.eval()` in `before_epoch()` ?

Same would hold for the other `Segmentor/Classifier` impls ?
